### PR TITLE
Refs task #387261 Search User

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets

- [#387261](https://dev.sun-asterisk.com/issues/387261)

## What's this PR do ?

- [x] Remove datatable and use pagination of laravel instead
- [x] Implement search user

## Library
*(List package, library third party add new include version)*

- N/A

## Performance

- [ ] Resolved n + 1 query
- [ ] Run explain query already
- [ ] Time run rake task : 1000 ms

## Checklist

- [x] It was tested in local success?
- [x] Fill link PR into ticket and the opposite
- [x] Note purpose/reason, solution, scope of influence into ticket

## ENV note

```
```

## Notes
*(Other notes)*
image

### CODE REVIEW

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>failure</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr></table>

File: app/Repositories/Eloquents/UserRepository.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+class UserRepository extends BaseRepository implements UserRepositoryInterface
+{
+    public function getModel()
+    {
+        return User::class;
+    }
+
+    public function getAllAndSearch($perPage, $searchParams)
+    {
+        $query = $this->model->orderBy('created_at', 'desc');
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/UserRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.ControlStructureSpacing.SpacingAfterOpen</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/UserRepository.php</br>**Line**: 21</br>Blank line found at start of control structure

File: app/Repositories/Eloquents/UserRepository.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+        if ($searchParams) {
+
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/UserRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.ControlStructureSpacing.SpacingAfterOpen</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/UserRepository.php</br>**Line**: 24</br>Blank line found at start of control structure

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr></table>

**Review by quanghung97 — APPROVED** [review]
OK

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr></table>

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>failure</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr></table>

File: app/Repositories/Eloquents/UserRepository.php [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+use App\Repositories\Eloquents\BaseRepository;
+use App\Repositories\Interfaces\UserRepositoryInterface;
+
+class UserRepository extends BaseRepository implements UserRepositoryInterface
+{
+    public function getModel()
+    {
+        return User::class;
+    }
+
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/UserRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.FunctionOpeningBraceSpace.SpacingAfter</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/UserRepository.php</br>**Line**: 18</br>Expected 0 blank lines after opening function brace; 1 found

File: app/Repositories/Eloquents/UserRepository.php [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+            return $query->search($searchParams, $perPage);
+        } else {
+            return $query->paginate($perPage);
+        }
+    }
+
+    private function handleSearchParams($searchParams)
+    {
+        $searchKeys = ['name', 'email'];
+        foreach ($searchKeys as $searchKey) {
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/UserRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/UserRepository.php</br>**Line**: 35</br>String "" does not require double quotes; use single quotes instead

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>failure</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr></table>

File: app/Repositories/Eloquents/UserRepository.php [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/UserRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/UserRepository.php</br>**Line**: 35</br>String "" does not require double quotes; use single quotes instead

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>php_artisan_migrate_._start_run_.19</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.15</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>yarn_dev._start_run_.9</td><td>success</td><td></td></tr><tr><td>composer_dump-autolo._start_run_.17</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>yarn_install._start_run_.7</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.11</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.13</td><td>success</td><td></td></tr></table>

File: resources/views/admins/users/index.blade.php [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+            <tbody>
+                @if(Request::get('search') && count($users) == 0)
+                    <p class="tbl-no-data "><i class="fa fa-info-circle"></i> No matching records found</p>
+                @elseif(count($users) == 0)
+                    <p class="tbl-no-data "><i class="fa fa-info-circle"></i> No data</p>
+                @else
+                    @foreach($users as $user)
+                        <tr>
+                            <td class="text-center">{{ $loop->iteration }}</td>
+                            <td class="text-center"><img src="img/placeholders/avatars/avatar12.jpg" alt="avatar" class="img-circle"></td>
```
Comment (tamtt-1031) [inline resources/views/admins/users/index.blade.php]: lấy avatar của user luôn em.
nếu ko có thì kiếm hình default bỏ vào. a thấy hình e chụp ko hiện hình lên tề

File: resources/views/admins/users/index.blade.php [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+            </thead>
+            <tbody>
+                @if(Request::get('search') && count($users) == 0)
+                    <p class="tbl-no-data "><i class="fa fa-info-circle"></i> No matching records found</p>
+                @elseif(count($users) == 0)
+                    <p class="tbl-no-data "><i class="fa fa-info-circle"></i> No data</p>
+                @else
+                    @foreach($users as $user)
+                        <tr>
+                            <td class="text-center">{{ $loop->iteration }}</td>
```
Comment (tamtt-1031) [inline resources/views/admins/users/index.blade.php]: trang thứ 2 đúng số thứ tự ko nhỉ :v

File: resources/views/admins/users/index.blade.php [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+</div>
+
+<div class="block full">
+    <div class="block-title">
+        <h2><strong>List user</strong></h2>
+    </div>
+    <div class="table-responsive">
+        <table class="table table-vcenter table-striped">
+            <thead>
+                <tr>
```
Comment (tamtt-1031) [inline resources/views/admins/users/index.blade.php]: a nghĩ No., Name, Email ko canh giữa sẽ đẹp hơn á

File: resources/views/admins/users/index.blade.php [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+                    <th class="text-center">No.</th>
```
Comment (tamtt-1031) [inline resources/views/admins/users/index.blade.php]: a nghĩ No., Name, Email ko canh giữa sẽ đẹp hơn á

**Review by nguyenthinhhn — APPROVED** [review]

**Review by tamtt-1031 — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Comment by phucqd-1305** [comment]
OK

**Pull request merged**